src/app/components/components.component.ts

```diff
@@ -1,14 +1,31 @@
 import {Component} from '@angular/core';
 import {DocItem, DocumentItemsService} from '../core/document-items.service';
+import {ActivatedRoute, NavigationEnd, Router} from '@angular/router';
 
 @Component({
     selector: 'hc-demo-components',
     templateUrl: './components.component.html'
 })
 export class ComponentsComponent {
     docItems: DocItem[];
+    thisPage = '';
+    selectOptions: Array<string> = [];
 
-    constructor(docItemService: DocumentItemsService) {
+    constructor(docItemService: DocumentItemsService, private activatedRoute: ActivatedRoute, private router: Router) {
         this.docItems = docItemService.getDocItems();
+
+        // Listen for vertical tab bar navigation and update the select component
+        router.events.subscribe(event => {
```

Comment:
You'll need to unsubscribe from the observable

Reply:
Thanks for walking me through this fix @jtrujill - let me know if the updates were implemented correctly